docs/source/reference-core.rst

```diff
@@ -802,9 +802,9 @@ to set the default behavior for any nursery in your program that doesn't overrid
   wrapping, so you'll get maximum compatibility with code that was written to
   support older versions of Trio.
 
-To maintain backwards compatibility, the default is ``strict_exception_groups=False``.
-The default will eventually change to ``True`` in a future version of Trio, once
-Python 3.11 and later versions are in wide use.
+The default is set to ``strict_exception_groups=True`` in line with the default behaviour
+of ``TaskGroup`` in asyncio and anyio. This is also to avoid any bugs caused by only
+catching one type of exceptions/exceptiongroups.
```

Comment:
I think that we'll eventually want to rewrite this whole section from the new strict-by-default perspective, under the heading "Deprecated: non-strict `ExceptionGroup`s" - to explain that it only exists for backwards-compatibility, will be removed in future, and that we recommend against it for all new code.

I don't want to delay this PR further though, so let's open an issue to revisit the docs on this topic rather than waiting to merge.